com.civicrm.hrjobroles/CRM/Hrjobroles/Upgrader.php

```diff
@@ -78,6 +78,85 @@ public function upgrade_1006() {
     return TRUE;
   }
 
+  /**
+   * Updates job role having wrong funder id with correct contact id.
+   *
+   * The wrong funder ids were as a result of contact api not returning
+   * correct contact id. When requesting for list of individual contact
+   * types to be used as funders, the api returns job role id instead of
+   * contact id whenever a contacts have existing job role.
+   *
+   * To decide the correct id, the funders in each job roles are checked
+   * against contacts whose contact types are individual. The id of contacts,
+   * whose job role id matches the funder are used to update the previously
+   * saved funder.
+   *
+   * @return bool
+   * @throws CiviCRM_API3_Exception
+   */
+  public function upgrade_1007() {
+    $contactDetails = $this->getContactsWithJobRole();
+    $jobRoles = civicrm_api3('HrJobRoles', 'get', [
+      'funder' => ['!=' => '|'],
+    ]);
+
+    foreach($jobRoles['values'] as $key => $value) {
+      $funders = explode('|', $value['funder']);
+      $funders = array_filter($funders);
+
+      list($funders, $allowUpdate) = $this->changeFunderToCorrectContactId($funders, $contactDetails);
+      if ($allowUpdate) {
+        civicrm_api3('HrJobRoles', 'create', [
+          'id' => $key,
+          'funder' => '|' . implode('|', $funders) . '|',
+        ]);
+      }
+    }
+
+    return TRUE;
+  }
+
+  /**
+   * Retrieves the correct contact for a funder and
+   * indicates update is required when found
+   *
+   * @param array $funders
+   * @param array $contactDetails
+   *
+   * @return array
+   */
+  private function changeFunderToCorrectContactId($funders, $contactDetails) {
+    $allowUpdate = FALSE;
+    foreach ($funders as $index => $funder) {
+      $jobRoleIds = array_column($contactDetails, 'jobrole_id');
+      $funderIndex = array_search($funder, $jobRoleIds, TRUE);
```

Comment:
Could you please give more details about why you're doing a strict search? My concern here is that funder is a string (because it's one item from the array created with an explode), while `$jobRoleIds` is more likely to be an array of integers, which will fail in a strict comparison.

Reply:
Using strict is not compulsory here actually. This was suggested when i used php inspection tool. Secondly, the `$jobRoleIds` were represented as array of strings when i tested it. I think its better to remove strict check 🤷‍♂️. What would you advice?

Reply:
Remove it